modules/mtl/arrow-mtl/src/main/kotlin/arrow/mtl/extensions/statet.kt

```diff
@@ -293,3 +296,21 @@ interface StateTAlternative<F, S> : Alternative<StateTPartialOf<F, S>>, StateTMo
   override fun <A> StateTOf<F, S, A>.combineK(y: StateTOf<F, S, A>): StateT<F, S, A> =
     orElse(y).fix()
 }
+
+@extension
+interface StateTEqK<F, S> : EqK<StateTPartialOf<F, S>> {
+  fun EQKF(): EqK<F>
+  fun EQS(): Eq<S>
+  fun M(): Monad<F>
+  fun s(): S
+
+  override fun <A> Kind<StateTPartialOf<F, S>, A>.eqK(other: Kind<StateTPartialOf<F, S>, A>, EQ: Eq<A>): Boolean =
+    (this.fix() to other.fix()).let {
+      val ls = it.first.runM(M(), s())
+      val rs = it.second.runM(M(), s())
+
+      EQKF().liftEq(Tuple2.eq(EQS(), EQ)).run {
+        ls.eqv(rs)
+      }
+    }
+}
```

Comment:
On second thought, please only use this in the tests. This is another case of checking function equality and thus this won't ever be accurate, not something we should have on the public api.